tools/compliance/write_sbom.py

```diff
@@ -67,21 +70,40 @@ def create_sbom(package_info: dict) -> dict:  # pylint: disable=g-bare-generic
   })
 
   for pkg in package_info["packages"]:
-    packages.append(
-        {
-            "name": pkg,
-            # TODO(aiuto): Fill in the rest
-            # "SPDXID": "SPDXRef-GooglePackage-4c7dc29872b9c418",
-            # "supplier": "Organization: Google LLC",
-            # "downloadLocation": "NOASSERTION",
-            # "licenseConcluded": "License-da09db95a268defe",
-            # "copyrightText": ""
-        }
-    )
+    id = hashlib.md5()
+    id.update(pkg.encode('utf-8'))
+    spdxid = 'SPDXRef-GooglePackage-%s' % id.hexdigest()
+    pi = {
+        "name": pkg,
+        "downloadLocation": "NOASSERTION",
+        "SPDXID": spdxid,
+        # TODO(aiuto): Fill in the rest
+        # "supplier": "Organization: Google LLC",
+        # "licenseConcluded": "License-da09db95a268defe",
+        # "copyrightText": ""
+    }
+
+    have_maven = None
+    if pkg.startswith('@maven//:'):
+      have_maven = maven_packages.get(pkg[9:])
+    elif pkg.endswith("//file:file"):
+      # Bazel hacks jvm_external to add //file:file as a target, then we depend
+      # on that rather than the correct thing.
+      # Example: @org_apache_tomcat_tomcat_annotations_api_8_0_5//file:file
+      # Check for just the versioned root
+      have_maven = maven_packages.get(pkg[1:-11])
```

Comment:
Python style is usually to do something like -len("//file:file") rather than have -11 as a magic number.

Reply:
I moved the string out to a constant.